src/core/lib/iomgr/error.cc

```diff
@@ -982,4 +982,17 @@ bool grpc_log_error(const char* what, grpc_error_handle error, const char* file,
   return false;
 }
 
+namespace grpc_core {
+void AddFieldError(absl::string_view field_name, absl::string_view error,
+                   std::vector<grpc_error_handle>* error_list) {
+  AddStringToErrorList(absl::StrCat("field:", field_name, " error:", error),
+                       error_list);
+}
+
+void AddStringToErrorList(absl::string_view error,
+                          std::vector<grpc_error_handle>* error_list) {
+  error_list->push_back(GRPC_ERROR_CREATE_FROM_CPP_STRING(std::string(error)));
```

Comment:
Moving the call to `GRPC_ERROR_CREATE_FROM_CPP_STRING()` from the call site into here means that the resulting `grpc_error` objects will encode the wrong file and line number, which sort of defeats the purpose of using `grpc_error` for JSON parsing errors in the first place.

I think we could live without the file and line numbers for JSON parsing, but if we're going to do that, then maybe we should switch to just storing the errors as simple strings instead of creating a `grpc_error` for each one.

Reply:
I think we should probably provide them. Next time I'm at a computer I'll look at what putting a macro in for this looks like.

Reply:
(there's still a lot of callsites where we do fancier composition of errors so dropping using error seems impractical)

Reply:
Thinking about this further, I suspect that we may actually be better off not using `grpc_error` for the individual errors within JSON parsing.

In general, when parsing a JSON object, the goal is to report *all* errors in all fields to the user, because it would be much harder to use if we just stopped at the first error and required the user to iterate after fixing each individual error.  Including all errors means that the error message needs to be somewhat structured, so that the user can see which error is in what part of the object hierarchy.  At the time we first started doing this, we weren't able to use STL or absl, so it was very hard to construct and combine strings in an effective way to provide that kind of structured error message.  So we wound up using the ability to store a tree of `grpc_error` objects as a way of providing that structure.  But the result of this actually has so much noise that it's not really that friendly to users: they really don't care what file or line number in our code produced the error (or the timestamp or any other `grpc_error` metadata); all they really care about where in JSON tree the error occurred.

I think now that we can use STL and absl, we have the tools we need to provide only what the user actually wants here.  Instead of using vectors of `grpc_error`, we could use vectors of strings, and we could easily combine them into parent error messages with `absl::StrJoin()`.